Source/WebCore/svg/SVGElement.cpp

```diff
@@ -925,8 +833,152 @@ CSSPropertyID SVGElement::cssPropertyIdForSVGAttributeName(const QualifiedName&
     if (!attrName.namespaceURI().isNull())
         return CSSPropertyInvalid;
 
-    static NeverDestroyed properties = createAttributeNameToCSSPropertyIDMap();
-    return properties.get().get(attrName.localName());
+    switch (attrName.nodeName()) {
```

Comment:
Why not use a `SortedArrayMap`? It was designed for this, and could possibly be more efficient than a switch statement. Maybe you tried it and it was less efficient? Or maybe it's hard to keep the list in numeric order?

Reply:
In the past whenever I tried to replace something with a SortedArrayMap in a hot code path, it ended up being slightly slower. I haven't tried this particular case though. Do we really think SortedArrayMap (with linear search) could be faster than a switch statement?

I'll find some time to try it out but I am not super hopeful.